aws_lambda_builders/workflows/nodejs_npm_esbuild/esbuild.py

```diff
@@ -102,3 +95,198 @@ def run(self, args, cwd=None):
             raise EsbuildExecutionError(message=err.decode("utf8").strip())
 
         return out.decode("utf8").strip()
+
+
+# The esbuild API flags are broken up into three forms (https://esbuild.github.io/api/):
+# Boolean types (--minify)
+SUPPORTED_ESBUILD_APIS_BOOLEAN = [
+    "minify",
+    "sourcemap",
+]
+
+# single value types (--target=es2020)
+SUPPORTED_ESBUILD_APIS_SINGLE_VALUE = [
+    "target",
+]
+
+# Multi-value types (--external:axios --external:aws-sdk)
+SUPPORTED_ESBUILD_APIS_MULTI_VALUE = [
+    "external",
+    "loader",
+]
+
+
+class EsbuildCommandBuilder:
+    ENTRY_POINTS = "entry_points"
+
+    def __init__(
+        self, scratch_dir: str, artifacts_dir: str, bundler_config: Dict[Any, Any], osutils: OSUtils, manifest: str
+    ):
+        self._scratch_dir = scratch_dir
+        self._artifacts_dir = artifacts_dir
+        self._bundler_config = bundler_config
+        self._osutils = osutils
+        self._manifest = manifest
+        self._command = []
```

Comment:
nit: I assume this is `List[str]` should we add typing to this since it is empty list when we initialize this?